Tests/AnyCodableTests/KeyedEncodingContainerProtocol+AnyCodableTests.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  KeyedEncodingContainerProtocol+AnyCodableTests.swift
+//  AnyCodableTests
+//
+//  Created by Dale Buckley on 22/03/2018.
+//
+
+import XCTest
+@testable import AnyCodable
+
+class KeyedEncodingContainerProtocolAnyCodableTests: XCTestCase {
+    
+    static let allTests = [
+        ("testEncodingEmptyAnyCodable", testEncodingEmptyAnyCodable),
+        ("testEncodingNonEmptyAnyCodable", testEncodingNonEmptyAnyCodable)
+    ]
+
+    func testEncodingEmptyAnyCodable() throws {
+
+        let mock = MockCodable(nil)
+        let data = try JSONEncoder().encode(mock)
+
+        XCTAssertFalse(String(data: data, encoding: .utf8)!.contains("\"any\":"))
+
+        let decoded = try JSONDecoder().decode(MockCodable.self, from: data)
+
+        XCTAssertNil(decoded.any)
+    }
+
+    func testEncodingNonEmptyAnyCodable() throws {
+
+        let mock = MockCodable("test")
+        let data = try JSONEncoder().encode(mock)
+
+        XCTAssertTrue(String(data: data, encoding: .utf8)!.contains("\"any\":"))
+
+        let decoded = try JSONDecoder().decode(MockCodable.self, from: data)
+
+        XCTAssertEqual(decoded.any as! String, "test")
+    }
+}
+
+extension KeyedEncodingContainerProtocolAnyCodableTests {
+
+    struct MockCodable: Codable {
```

Comment:
why not just
```
struct MockCodale: Codable {
  let name: String
}
```

Reply:
Well that particular case will just use the `String` overload. If we want to see if auto synth will work then it should have a property of `AnyCodable`.